src/views/DashboardView.vue

```diff
@@ -0,0 +1,156 @@
+<template>
+  <div class="main-container">
+    <boards-navbar
+      v-bind="boardsNavbarProps"
+      :class="{
+        'absolute sm:scale-0': windowWidth < 640,
+        'scale-0 sm:scale-100': windowWidth >= 640
+      }"
+    />
+
+    <main-navbar
+      :sidebar="isSidebarShown"
+      :isLogo="isLogoShown"
+      :theme="isDark"
+      :dashboard="isDashboardEmpty"
+      :width="windowWidth"
+      :boardName="boardProperties.boardName"
+      :areOptionsShown="areBoardOptionsShown"
+      :navOpen="isNavOpen"
+      :toggleOptions="($event) => callMoreOptionsFn($event, toggleOptions)"
+      :closeOptions="($event) => callMoreOptionsFn($event, closeOptions)"
+      :toggleBoardsNav="toggleBoardsNav"
+      :addTask="() => toggleDialog('addTask')"
+      :editTask="() => toggleDialog('editTask')"
+      :editBoard="() => toggleDialog('editBoard')"
+      :deleteTask="() => toggleDialog('deleteTask')"
+      :deleteBoard="() => toggleDialog('deleteBoard')"
+      :class="{ 'col-span-2': isLogoShown }"
+    />
+
+    <div
+      @click="toggleSidebar"
+      v-show="isLogoShown && windowWidth >= 640"
+      class="show-sidebar">
+      <img src="/img/icon-show-sidebar.svg" alt="show sidebar">
+    </div>
+
+    <main
+      class="flex flex-col justify-center p-4 sm:p-6"
+      :class="{ 'sm:col-start-2': !isLogoShown, 'sm:col-start-1 sm:col-span-2': isLogoShown }"
+    > 
+      <empty-info />
+
+      <boards-column
+        :columns="boardProperties.columns"
+        :logo="isLogoShown"
+        :callback="() => toggleDialog('seeTask')"
+      />
+
+      <dialogs-template
+        v-bind="dialogProps"
+      />
+    </main>
+  </div>
+</template>
+
+<script setup lang="ts">
+import type { SharedProps, ToggledDialog, PropsToSelect } from '../api/dialogTypes'
+import MainNavbar from '../components/Navbar/MainNavbar.vue'
+import BoardsNavbar from '../components/Navbar/BoardsNavbar.vue'
+import EmptyInfo from '../components/EmptyInfo.vue'
+import BoardsColumn from '../components/BoardsColumn.vue'
+import DialogsTemplate from '../components/Dialogs/DialogsTemplate.vue'
+import { returnBoardProperties } from '../composables/boardProperties'
+import { returnDialogPropsToSelect } from '../composables/dialogHandling'
+import moreOptionsPopup from '../composables/moreOptionsPopup'
+import { ref, Ref, computed } from 'vue'
+import { useDark } from '@vueuse/core'
+
+const isDark = useDark()
+const isDashboardEmpty = ref(false)
+const boardProperties = returnBoardProperties()
+
+const areBoardOptionsShown = ref(false)
+const boardsNavbarProps = computed(() => ({
+  condition: windowWidth.value < 640 ? isNavOpen.value : isSidebarShown.value,
+  width: windowWidth.value,
+  theme: isDark.value,
+  boards: boardProperties.boards,
+  boardName: boardProperties.boardName,
+  toggleSidebar: toggleSidebar,
+  addNewBoard: () => toggleDialog('addBoard')
+}))
```

Comment:
Will viewport resize trigger a rerender?

Reply:
I'd say yes, since I used `resizeObserver` for observing the innerWidth of the window, like so:
```js
const windowWidth = ref(window.innerWidth)
const resizeObserwer = new ResizeObserver((entries) => {
  windowWidth.value = entries[0].contentRect.width
})
resizeObserwer.observe(document.body)
```

And as this `boardsNavbarProps` is a computed property, it'll "update" everytime any of its "dependencies" changes.

Reply:
use this instead - https://vueuse.org/core/useWindowSize/#usewindowsize